gtests/test_nopython_callprofilerserializer.cpp

```diff
@@ -0,0 +1,554 @@
+            {
+                "unique_number": 0,
+                "key": 0,
+                "name": "void modmesh::detail::func1()",
+                "data": {
+                    "start_time": 2074704984391166,
+                    "caller_name": "void modmesh::detail::func1()",
+                    "total_time": 61000834,
+                    "call_count": 1,
+                    "is_running": 1
+                },
+                children": [
+                    3
+                ]
+            },
+            {
+                "unique_number": 1,
+                "key": 1,
+                "name": "void modmesh::detail::func2()",
+                "data": {
+                    "start_time": 2074705045392250,
+                    "caller_name": "void modmesh::detail::func2()",
+                    "total_time": 54001000,
+                    "call_count": 1,
+                    "is_running": 1
+                },
+                children": [
+                    4
+                ]
+            },
+            {
+                "unique_number": 2,
+                "key": 2,
+                "name": "void modmesh::detail::func3()",
+                "data": {
+                    "start_time": 2074705118393708,
+                    "caller_name": "void modmesh::detail::func3()",
+                    "total_time": 38000208,
+                    "call_count": 2,
+                    "is_running": 1
+                },
+                children": []
+            },
+            {
+                "unique_number": 3,
+                "key": 1,
+                "name": "void modmesh::detail::func2()",
+                "data": {
+                    "start_time": 2074704984391458,
+                    "caller_name": "void modmesh::detail::func2()",
+                    "total_time": 54000417,
+                    "call_count": 1,
+                    "is_running": 1
+                },
+                children": [
+                    5
+                ]
+            },
+            {
+                "unique_number": 4,
+                "key": 2,
+                "name": "void modmesh::detail::func3()",
+                "data": {
+                    "start_time": 2074705080392791,
+                    "caller_name": "void modmesh::detail::func3()",
+                    "total_time": 19000417,
+                    "call_count": 1,
+                    "is_running": 1
+                },
+                children": []
+            },
+            {
+                "unique_number": 5,
+                "key": 2,
+                "name": "void modmesh::detail::func3()",
+                "data": {
+                    "start_time": 2074705019391625,
+                    "caller_name": "void modmesh::detail::func3()",
+                    "total_time": 19000125,
+                    "call_count": 1,
+                    "is_running": 1
+                },
+                children": []
+            }
+        ]
+    }
+}
+    */
```

Comment:
I'm not sure whether I should show the output in the comment.

Reply:
At the time being let's keep it here and if it's no good we can remove it later.

It's strange that I cannot see some of your comments like this inline.  Github PR code review is really bad.

Reply:
> At the time being let's keep it here and if it's no good we can remove it later.
> 
> It's strange that I cannot see some of your comments like this inline. Github PR code review is really bad.

I don't get it. 
Do you mean that I should add more comment based on the example in the comment?